crates/gitbutler-core/src/git/error.rs

```diff
@@ -49,6 +52,12 @@ impl From<git2::Error> for Error {
     }
 }
 
+impl ErrorWithContext for Error {
```

Comment:
An empty implementation shouldn't be required. It's interesting you found the need anyway.

Reply:
Thanks for checking my work 🙌 

The reason I added that was because it felt cumbersome to manually convert context-free `git:Error` using `Error::from_err`, as mentioned in this comment.
https://github.com/gitbutlerapp/gitbutler/blob/f6421aab3c9ea36944fce6d690602ef2586280f6/crates/gitbutler-core/src/error.rs#L112-L121

Reply:
Ah, I see, thank you. It's the auto-conversion you were after.
It likely that there is no way around it though, but I hope that now using `thiserror` most of the time is the solution, ultimately.
I will watch out for this when trying to refactor error handling.